wdl/MasterVcfQc.wdl

```diff
@@ -34,6 +34,7 @@ workflow MasterVcfQc {
     Array[File]? custom_comparison_sets
     Array[String] contigs
     Int? random_seed
+    File? vcf_idx
```

Comment:
I'm not sure if you discussed this with @mwalker174 or not, but we recently removed vcf indices as explicit inputs to subworkflows in the pipeline, see https://github.com/talkowski-lab/gatk-sv-v1/pull/252.

Is the reason for adding it back the fact that this module is sometimes called without an index file having been created? Is there any chance we could just remedy that situation rather than reverting to passing the index in this particular case? In my opinion we should be creating indices alongside vcf files consistently through the pipeline.

Reply:
The vcf cannot be tabix-indexed at this point in the pipeline because it is out of spec - the END field is used anomalously for BNDs so END < START sometimes, and that is not fixed until the very end of the pipeline, and Mark felt it was easier to do a workaround here rather than change everything in between to get it to comply with vcf spec. That means MasterVcfQc.wdl needs to be called with an index for Module05_06 and without for Module03Qc, which motivated this change. There is still no need to provide a VCF index file path on the top-level WDL though. But also, if you have suggestions for a better way to accomplish this I would welcome that - I originally wanted to just check if the expected .tbi file existed but could not figure out how to do that in WDL.

Reply:
Oh, sorry, you mentioned the END thing before and I forgot.

One potential option to check if the tbi file exists might be to use the WDL `size()` function on it. It looks like if you pass it a `File?` and the file doesn't exist it should return 0:

https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#float-sizefile-string

I haven't tested that myself, though, and it's a little messy looking.

Reply:
It looks to me like the behavior if the file is provided as an argument to `size()` but does not exist is not specified in the docs - I read it as returning 0 when no File argument is provided. Since it's a bit unclear and the current implementation still avoids having to provide `vcf_idx` on the top level WDL, I'm inclined to leave this as-is - but could also test out `size()` and see what happens if you think that's a better way

Reply:
That's fine, it's not a big deal either way.